From c2d79b1854f7415e7eb9bc5f2b3a332aed1bcf12 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Sun, 10 Aug 2014 18:20:29 -0400
Subject: [iommu] Change iommu driver to call remove_device_to_group trace event

Message-id: <20140810182029.24055.73156.stgit@gir.stowe>
Patchwork-id: 87581
O-Subject: [RHEL7.1 PATCH 10/19] iommu: Change iommu driver to call remove_device_to_group trace event
Bugzilla: 1127467
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Dutile <ddutile@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1127467
Upstream Status: 2e757086bdfdc9450dc2e4a5d2ec5431520a02c8

commit 2e757086bdfdc9450dc2e4a5d2ec5431520a02c8
Author: Shuah Khan <shuah.kh@samsung.com>
Date:   Thu Aug 15 11:59:25 2013 -0600

    iommu: Change iommu driver to call remove_device_to_group trace event

    Change iommu driver to call remove_device_to_group trace event. This
    iommu_group class event can be enabled to trigger when devices get
    removed from an iommu group. Trace information includes iommu group id and
    device name.

    Testing:
    Added trace calls to iommu_prepare_identity_map() for testing some of the
    conditions that are hard to trigger. Here is the trace from the testing:

           swapper/0-1     [003] ....     1.854101: remove_device_from_group: IO

    Signed-off-by: Shuah Khan <shuah.kh@samsung.com>
    Signed-off-by: Joerg Roedel <joro@8bytes.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/iommu/iommu.c b/drivers/iommu/iommu.c
index 349c92d..278055b 100644
--- a/drivers/iommu/iommu.c
+++ b/drivers/iommu/iommu.c
@@ -402,6 +402,8 @@ void iommu_group_remove_device(struct device *dev)
  sysfs_remove_link(group->devices_kobj, device->name);
  sysfs_remove_link(&dev->kobj, "iommu_group");
 
+ trace_remove_device_from_group(group->id, dev);
+
  kfree(device->name);
  kfree(device);
  dev->iommu_group = NULL;
-- 
1.7.1